roles/acme/tasks/challenge/dns-01/autodns.yml

```diff
@@ -5,37 +5,24 @@
   block:
     - name: Login to autodns # noqa no-handler
       ansible.builtin.uri:
-        url: "https://api.autodns.com/v1/login"
+        url: https://api.autodns.com/v1/login
         method: POST
         body_format: json
-        body:
-          { "context": "4",
-            "user": "{{ acme_dns_user }}",
-            "password": "{{ acme_dns_password }}"
-          }
+        body: { context: "4", user: "{{ acme_dns_user }}", password: "{{ acme_dns_password }}" }
       register: login
 
     - name: Add a new TXT record to the SAN domains
       ansible.builtin.uri:
-        url: "https://api.autodns.com/v1/zone/{{ acme_domain.zone }}/a.ns14.net"
+        url: https://api.autodns.com/v1/zone/{{ acme_domain.zone }}/a.ns14.net
         method: PATCH
         body_format: json
         body:
-          {
-            "origin": "{{ item }}",
-            "resourceRecordsAdd": [
-              {
-                # "remove" '*.' from entry when it occurs. without it it would try to create a
-                # record like _acme-challenge.*.example.com which is not allowed by AutoDNS
-                # see: https://jinja.palletsprojects.com/en/master/templates/#replace
-                # and: https://jbmoelker.github.io/jinja-compat-tests/filters/replace/#pattern
-                "name": "_acme-challenge.{{ item | replace('*.', '') }}",
-                "ttl": 60,
-                "type": "TXT",
-                "value": "{{ challenge['challenge_data'][item]['dns-01']['resource_value'] }}"
-              }
-            ]
-          }
+          origin: "{{ item }}"
+          resourceRecordsAdd:
+            - name: _acme-challenge.{{ item | replace('*.', '') }}
+              ttl: 60
+              type: TXT
+              value: "{{ challenge['challenge_data'][item]['dns-01']['resource_value'] }}"
```

Comment:
Question for all changes in this regard:
This PR translates some JSON to YAML. In this context I really like the JSON formatting, since it is a API payload and I find it easier to reason about it, when this stays JSON.

Reply:
I don't really care. I'd jsut have to make sure that `ansible-lint --write` won't change it again. @avalor1 what do you think?

Reply:
I'm with schurzi on this one. As most of the tools one uses are doing it like that, I also prefer the json formatting, but i can see that the yaml formatting looks very appealing as it looks a little cleaner.